portality/upgrade.py

```diff
@@ -94,13 +98,24 @@ def do_upgrade(definition, verbose):
 
             # When we have enough, do some writing
             if len(batch) >= batch_size:
-                print "writing ", len(batch), "to", tdef.get("type")
+                total += len(batch)
+                print "writing ", len(batch), "to", tdef.get("type"), ";", total, "of", max
                 esprit.raw.bulk(tconn, batch, type_=tdef.get("type"))
                 batch = []
+                # do some timing predictions
+                batch_tick = datetime.now()
+                time_so_far = batch_tick - type_start
+                seconds_so_far = time_so_far.total_seconds()
+                estimated_seconds_remaining = ((seconds_so_far * max) / total) - seconds_so_far
```

Comment:
why isn't this `((seconds_so_far / total) * max) - seconds_so_far`?

i.e. you know how many you've done, you estimate how long it's taken per item and guess the full batch time from that.

Reply:
They are mathematically equivalent :)